### PR TITLE
Portage/GHCCore.hs: add ghc-9.0.1 library detection

### PR DESCRIPTION
**Pull request by l29ah**

### CODE REVIEW

File: Portage/GHCCore.hs [hunk 1]
```diff
@@ -167,6 +170,42 @@ ghc843 = (ghc [8,4,3], mkIndex ghc843_pkgs)
 --      example: https://downloads.haskell.org/~ghc/8.6.5/docs/html/users_guide/8.6.5-notes.html
 --  * our binary tarballs (package.conf.d.initial subdir)
 --  * ancient: http://haskell.org/haskellwiki/Libraries_released_with_GHC
+ghc901_pkgs :: [Cabal.PackageIdentifier]
+ghc901_pkgs =
+--  , p "Cabal" [3,2,0,0]  package is upgradeable
```
Comment (trofi) [inline Portage/GHCCore.hs:175]: Wrong cabal version in the comment, should be `3.4.0.0`.

File: Portage/GHCCore.hs [hunk 2]
```diff
@@ -167,6 +170,42 @@ ghc843 = (ghc [8,4,3], mkIndex ghc843_pkgs)
...
+  [ p "array" [0,5,4,0]
+  , p "base" [4,15,0,0]
+  , p "binary" [0,8,8,0] -- used by libghc
+  , p "bytestring" [0,10,12,1]
+  , p "containers" [0,6,4,1]
+  , p "deepseq" [1,4,5,0] -- used by time
+  , p "directory" [1,3,6,1]
+--  , p "exceptions [0,10,4] -- used by haskeline, package is upgradeable 
+  , p "filepath" [1,4,2,1]
+  , p "ghc-boot-th" [9,0,1]
```
Comment (trofi) [inline Portage/GHCCore.hs:185]: `ghc-bignum` should be added here.

File: Portage/GHCCore.hs [hunk 3]
```diff
@@ -167,6 +170,42 @@ ghc843 = (ghc [8,4,3], mkIndex ghc843_pkgs)
...
+  , p "ghc-boot" [9,0,1]
+  , p "ghc-compact" [0,1,0,0]
+  , p "ghc-heap" [9,0,1]
+  , p "ghc-prim" [0,7,0]
+  , p "ghci" [9,0,1]
+--  , p "haskeline" [0,8,0,0]  package is upgradeable
```
Comment (trofi) [inline Portage/GHCCore.hs:191]: haskeline version is wrong in the comment, should be `0.8.1.0`

File: Portage/GHCCore.hs [hunk 4]
```diff
@@ -167,6 +170,42 @@ ghc843 = (ghc [8,4,3], mkIndex ghc843_pkgs)
...
+ghc901_pkgs =
+--  , p "Cabal" [3,2,0,0]  package is upgradeable
+  [ p "array" [0,5,4,0]
+  , p "base" [4,15,0,0]
+  , p "binary" [0,8,8,0] -- used by libghc
+  , p "bytestring" [0,10,12,1]
+  , p "containers" [0,6,4,1]
+  , p "deepseq" [1,4,5,0] -- used by time
+  , p "directory" [1,3,6,1]
+--  , p "exceptions [0,10,4] -- used by haskeline, package is upgradeable
```
Comment (trofi) [inline Portage/GHCCore.hs:183]: Looks like `exceptions` is used by `ghc` library itself. That effectively makes it unupgradeable.

File: Portage/GHCCore.hs [hunk 5]
```diff
@@ -167,6 +170,42 @@ ghc843 = (ghc [8,4,3], mkIndex ghc843_pkgs)
...
+  , p "ghc-boot" [9,0,1]
+  , p "ghc-compact" [0,1,0,0]
+  , p "ghc-heap" [9,0,1]
+  , p "ghc-prim" [0,7,0]
+  , p "ghci" [9,0,1]
+--  , p "haskeline" [0,8,0,0]  package is upgradeable
+  , p "hpc" [0,6,1,0] -- used by libghc
+  , p "integer-gmp" [1,1]
+  , p "libiserv" [9,0,1]
+  --  , p "mtl" [2,2,2]  package is upgradeable(?)
```
Comment (trofi) [inline Portage/GHCCore.hs:195]: `mtl` and `parsec` are used by `exceptions` (used by `ghc` library itself). That effectively makes them unupgradeable.

File: Portage/GHCCore.hs [hunk 6]
```diff
@@ -167,6 +170,42 @@ ghc843 = (ghc [8,4,3], mkIndex ghc843_pkgs)
...
+  --  , p "parsec" [3,1,14,0]  package is upgradeable(?)
+  , p "pretty" [1,1,3,6]
+  , p "process" [1,6,11,0]
+  --  , p "stm" [2,5,0,0]  package is upgradeable(?)
+  , p "template-haskell" [2,17,0,0] -- used by libghc
+  -- , p "terminfo" [0,4,1,4]
```
Comment (trofi) [inline Portage/GHCCore.hs:201]: `terminfo` is used by `ghc`. We'll need to mark it unupgradeable.

File: Portage/GHCCore.hs [hunk 7]
```diff
@@ -167,6 +170,42 @@ ghc843 = (ghc [8,4,3], mkIndex ghc843_pkgs)
...
+  -- , p "text" [1,2,3,2] dependency of Cabal library
```
Comment (trofi) [inline Portage/GHCCore.hs:202]: `text` is used by `parsec`. We would have to make it unupgradeable. Also, it's version is `1.2.4.1`.

File: Portage/GHCCore.hs [hunk 8]
```diff
@@ -140,6 +140,9 @@ ghc :: [Int] -> DC.CompilerInfo
 ghc nrs = DC.unknownCompilerInfo c_id DC.NoAbiTag
     where c_id = CompilerId GHC (mkVersion nrs)
 
+ghc901 :: (DC.CompilerInfo, InstalledPackageIndex)
```
Comment (trofi) [inline Portage/GHCCore.hs:143]: You'll also need to add `cabalFromGHC` entry for new `Cabal`.

**Review by trofi — CHANGES_REQUESTED** [review]

**Comment by hololeap** [comment]
This is all very outdated at this point. Reopen if you actually need this.

**Pull request closed without merging**